.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -2889,7 +2889,15 @@ stages:
             az storage blob upload --container-name "$(AZURE_STORAGE_CONTAINER_NAME)" --file "sha.txt" --name "sha.txt" --overwrite true
             az storage blob upload --container-name "$(AZURE_STORAGE_CONTAINER_NAME)" --file "version.txt" --name "version.txt" --overwrite true
           displayName: Upload indexes to Azure
-          condition: and(succeeded(), eq(variables.isMainBranch, true))
+          condition: >
+            and(
+              succeeded(),
+              ne(variables['push_artifacts_to_azure_storage'], 'false'),
+              or(
+                eq(variables['push_artifacts_to_azure_storage'], 'true'),
+                eq(variables.isMainBranch, true)
+              )
+            )
```

Comment:
The trouble with this is that we use these variables to ensure we're running the latest build in AAS for purposes of the release. Also note that this only uploads the indexes, you're not actually uploading the resources.

Instead, I think you should modify the condition on the Upload blobs to Azure step, and leave this one as it was

Reply:
you're right, thanks, that potentially saved me a lot of confusion about why stuff wasn't be updated!